src/components/EnvironmentDropdown/index.tsx

```diff
@@ -11,24 +10,14 @@ import {
   EnvironmentSelectorWrapper,
   FilterWrapper
 } from './style';
-
-type EnvironmentData = {
-  name: string;
-  meta: MetaDataProps['meta'];
-  testResult: {
-    passed: number;
-    failed: number;
-    skipped: number;
-  };
-};
-
-type EnvironmentDropdownProps = {
-  data: EnvironmentData[];
-};
+import { EnvironmentData, EnvironmentDropdownProps } from './types';
+import { filterData } from './utils';
 
 const EnvironmentDropdown: React.FC<EnvironmentDropdownProps> = ({ data }) => {
+  const [query, setQuery] = useState<string>('');
+  const filteredData = filterData(data, query);
   const [isDropDownOpen, setDropdownOpen] = useState(false);
-  const [context, setContext] = useState('Environment 2');
+  const [context, setContext] = useState(filteredData[0].name);
```

Comment:
filteredData contains an array of objects with the highest number of failures, so I am updating the state using the index. Can this be optimized?